src/ray/stats/metric_exporter.cc

```diff
@@ -25,10 +25,21 @@ void MetricExporter::ExportToPoints(
         &view_data,
     const std::string &metric_name, std::vector<std::string> &keys,
     std::vector<MetricPoint> &points) {
+  RAY_CHECK(is_exporting_ == false);
```

Comment:
Move these checker to upper level, I guess it's should be in MetricExporter::ExportViewData.
To make exporting and cleanup thread-safe, in my opinion, a spinlock could be good choice to do this. So code looks like

```cpp
MetricExporter::ExportViewData {
AutoSpinLock lock(exporting_flag);
if (StatsConfig::instance().IsInitialized()) {
  return;
}
// do some reporting thins.
}
```

Tiny spinlock is implemented by wrapping atomic bool flag in internal version.

Reply:
Exporting doesn't really have to be thread-safe. It is used only by a single thread (openCensus thread). 

Things that matter is just `is_exporting_` (to make sure if we have to wait until the current exporting phase is done) & `should_be_cleaned_up_` (which I think can be replaced by `IsInitialized`).